### ISSUE TITLE
False negative for pylint E0203 when using += operator

### ISSUE THREAD

**Issue by pylint-bot** [PyCQA/pylint#164]
Originally reported by: **Andreas Maier (BitBucket: [andi0661](http://bitbucket.org/andi0661))**

---

Hi,
I am using PyLint 1.1.0 with Python 2.6, and found that the following
code does not raise E0203 "Access to member '%s' before its
definition line %s", as it should:

```
 class MyClass1(object):
     def __init__(self):
         self.first += 5   # Does not raise E0203 as it should
         self.first = 0
```

Just for comparison, when omitting the straight assignment on the
second line of the method, message E1101 is raised, so the instance
attribute is (correctly) recognized to be read before the addition:

```
 class MyClass2(object):
     def __init__(self):
         self.first += 5   # Correctly raises E1101:
                           #   Instance of 'MyClass2' has no
                           #   'first' member
```

Also just for comparison, here is code that correctly raises E0203:

```
 class MyClass3(object):
     def __init__(self):
         self.first = self.sec  # Correctly raises E0203:
                                #   Access to member 'sec' before
                                #   its definition line <N>
         self.sec = 0
```

I did not verify whether or not the same issue exists for other x= operators.

Andy

---
- Bitbucket: https://bitbucket.org/logilab/pylint/issue/164
